src/System.Resources.Extensions/src/System/Resources/Extensions/PreserializedResourceWriter.cs

```diff
@@ -17,6 +22,9 @@ partial class PreserializedResourceWriter
         internal const string DeserializingResourceReaderFullyQualifiedName = "System.Resources.Extensions.DeserializingResourceReader, System.Resources.Extensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51";
         internal const string RuntimeResourceSetFullyQualifiedName = "System.Resources.Extensions.RuntimeResourceSet, System.Resources.Extensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51";
 
+        // an internal type name used to represent an unknown resource type 
+        private static string UnknownObjectTypeName = "System.Resources.Extensions.UnknownType, System.Resources.Extensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51";
```

Comment:
`const` instead of `static`?  Or could this be a `readonly static` and get the string dynamically rather than hardcoding it, e.g. `typeof(UnknownType).AssemblyQualifiedName`?

Reply:
I specifically do not want to get it dynamically because I don't want it to change when the assembly versions.

Reply:
Ok.  Then `static` => `const`?

Reply:
Yep

Reply:
Had a chat with @tarekgh and he made me think more about the desktop scenarios here.  A fixed type name that gets activated with older version on Core will work due to the binding policy, but not on desktop unless we have a bindingRedirect.  Most of the time we use this trick in lightup for framework assemblies, but that works because of unification.  Rather than rely on a bindingRedirect being present on desktop I think its better to use the full type name without assembly.  That will still succeed in activation since the caller will be the same assembly.  Alternatively I could use an invalid string here and change the type lookup code to avoid ever activating it, but that feels like more of a hack.  For now I will go with the sentinel and omit the assembly name.

Reply:
> I think its better to use the full type name without assembly

I agree with that and I think would satisfy our need.